src/parallel.ts

```diff
@@ -12,6 +12,16 @@ export interface IBuildTrackerSettings {
      */
     fileReader?: IFileReader;
 
+    /**
+     * When reading dependencies, also check optionalDependencies
+     */
+    includeOptionalDependencies?: boolean;
+
+    /**
+     * When reading dependencies, also check peerDependencies
+     */
+    includePeerDependencies?: boolean;
```

Comment:
A few thoughts:

* These are duplicated in `series.ts`. How about making an `IncludedDependencies` interface and re-using it?
* I think these should be opt-out by default with names like `exclude` to stay consistent with other types of dependencies.
* Are you up for also adding `devDependencies` to this list? If not I can file that as a followup issue, no worries.

Reply:
I refactored the interfaces to make a common `ISettings` interface that is made by composing smaller parts. Let me know if you find this expectable.

Also, I changed the options to be opt-out instead. If we ever need to switch them to something else we can change the defaults in the `normalizeSettings` function.

I added `excludeDevDependencies` as an option as well.

Reply:
Super, thanks!